packages/xstate-react/src/createSelector.ts

```diff
@@ -0,0 +1,12 @@
+import { StateFrom, StateMachine } from 'xstate';
+
+/**
+ * Allows you to specify a typesafe selector based on a machine definition.
+ *
+ * @example
+ * const getIsToggledOn = createSelector(machine, state => state.matches('toggledOn'));
+ */
+export const createSelector = <TMachine extends StateMachine<any, any, any>, T>(
+  _machine: TMachine,
+  selector: (state: TMachine['initialState']) => T
+) => selector;
```

Comment:
Is it possible to have `createSelector` generic enough for any actor? This is more of a `createMachineSelector` 🤔

Reply:
Yes true, it should probably use an ActorRef and grab the TEmitted

Reply:
@davidkpiano What would the first parameter take? Agree, this is more like a createMachineSelector. Not sure it can take an actor ref, because it needs to be created from the actor _specification_, not the actor. I.e. from the machine, not the running service.